io_scene_nif/modules/nif_import/property/shader/bsshaderproperty.py

```diff
@@ -102,85 +103,83 @@ def register(self, processor):
         processor.register(NifFormat.BSEffectShaderProperty, self.import_bs_effect_shader_property)
 
     def import_bs_lighting_shader_property(self, bs_shader_property):
-        b_mat = self.create_material_name(bs_shader_property)
 
         # Shader Flags
-        b_shader = b_mat.niftools_shader
+        b_shader = self.b_mat.niftools_shader
         b_shader.bs_shadertype = 'BSLightingShaderProperty'
 
         shader_type = NifFormat.BSLightingShaderPropertyShaderType._enumvalues.index(bs_shader_property.skyrim_shader_type)
         b_shader.bslsp_shaderobjtype = NifFormat.BSLightingShaderPropertyShaderType._enumkeys[shader_type]
 
-        self.import_shader_flags(b_mat, bs_shader_property)
+        self.import_shader_flags(bs_shader_property)
 
         # Textures
-        self.texturehelper.import_bsshaderproperty_textureset(b_mat, bs_shader_property)
-        if hasattr(bs_shader_property, 'texture_clamp_mode'):
-            self.import_clamp(b_mat, bs_shader_property)
-
-        if hasattr(bs_shader_property, 'uv_offset'):
-            self.import_uv_offset(b_mat, bs_shader_property)
-
-        if hasattr(bs_shader_property, 'uv_scale'):
-            self.import_uv_scale(b_mat, bs_shader_property)
+        self.texturehelper.import_bsshaderproperty_textureset(self.b_mat, bs_shader_property)
+        # if hasattr(bs_shader_property, 'texture_clamp_mode'):
+        #     self.import_clamp(self.b_mat, bs_shader_property)
+        #
+        # if hasattr(bs_shader_property, 'uv_offset'):
+        #     self.import_uv_offset(self.b_mat, bs_shader_property)
+        #
+        # if hasattr(bs_shader_property, 'uv_scale'):
+        #     self.import_uv_scale(self.b_mat, bs_shader_property)
 
         # Diffuse color
```

Comment:
I think we will have to redo this whole section as the original code is convoluted by itself but we should look to have specific type.

Reply:
Right, eventually we want to support the same for nitextureprops and use the shared material api class to take care of all the settings.